server/server.go

```diff
@@ -352,10 +332,13 @@ func (s *Server) Run() error {
 			}
 
 			logutil.BgLogger().Error("accept failed", zap.Error(err))
-			return errors.Trace(err)
+			return
 		}
 
 		clientConn := s.newConn(conn)
+		if listener == s.socket {
+			clientConn.isUnixSocket = true
```

Comment:
Maybe `startNetworkListener(xxx, isUnixSocket)`? I mean let the caller decide whether it is a unix socket.